src/generator/AutoRest.Go/Model/MethodGo.cs

```diff
@@ -245,12 +245,16 @@ public List<string> PrepareDecorators
               );
         }
 
-        if (RequestHeaders.Any())
+        if (HeaderParameters.Any())
         {
           foreach (var param in Parameters.Where(p => p.IsRequired && p.Location == ParameterLocation.Header))
           {
-            decorators.Add(string.Format("autorest.WithHeader(\"{0}\",autorest.String({1}))",
-               param.SerializedName, param.Name));
+            decorators.Add(string.Format(param.IsClientProperty
+                ? "autorest.WithHeader(\"{0}\",client.{1})"
+                : "autorest.WithHeader(\"{0}\",autorest.String({1}))",
+                    param.SerializedName, param.IsClientProperty
+                        ? param.Name.ToPascalCase().ToString()
+                        : param.Name.ToString()));
```

Comment:
Personally, I'm not a fan of nested ternary operators. How would you feel about refactoring to the following?

``` C#
    if (param.IsClientProperty) {
        decorators.Add($"autorest.WithHeader(\"{ param.SerializedName }\", client.{ param.Name.ToPascalCase().ToString() })");
    } else {
        decorators.Add($"autorest.WithHeader(\"{param.SerializedName}\", autorest.String({param.Name.ToString()}))");
    }
```